app/models/user.rb

```diff
@@ -110,6 +112,18 @@ def validate
   before_validation :dummy_password_for_external_auth
   before_destroy :destroy_subscribed_widget_sets
 
+  def check_reference
+    present_ref = []
+    %w[miq_requests vms miq_widgets miq_templates].each do |association|
```

Comment:
Should `miq_widgets` be removed from this list?  I think https://github.com/ManageIQ/manageiq/pull/18472/files#diff-4676c008b11a5480d73d4a6de01e45b9R113 will remove them, right?

Reply:
I think ```miq_widgets``` should stay since ```:destroy_subscribed_widget_sets``` will remove records from ```miq_sets``` table, not from ```miq_widgets```